Cannot build from source

At IzzyOnDroid we support [Reproducible Builds](https://reproducible-builds.org/) (see: [Reproducible Builds, special client support and more in our repo](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients)). Trying for yours, I was unable to even compile the APK:

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'repo'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find org.jetbrains.kotlinx:kotlinx-html-jvm:0.7.2.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/kotlinx/kotlinx-html-jvm/0.7.2/kotlinx-html-jvm-0.7.2.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/kotlinx/kotlinx-html-jvm/0.7.2/kotlinx-html-jvm-0.7.2.pom
       - https://plugins.gradle.org/m2/org/jetbrains/kotlinx/kotlinx-html-jvm/0.7.2/kotlinx-html-jvm-0.7.2.pom
     Required by:
         project : > com.jaredsburrows:gradle-license-plugin:0.8.90
```

Can you please tell me what I missed?

We'd appreciate if you could help making your build reproducible. We've prepared some [hints on reproducible builds](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Reproducible-Builds) for that.

Looking forward to your reply!

cc @obfusk

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -2,14 +2,14 @@ apply plugin: 'com.android.application'
 apply plugin: 'com.jaredsburrows.license'
 
 android {
-    compileSdkVersion 35
-    buildToolsVersion '35.0.0'
+    compileSdkVersion 33
+    buildToolsVersion '33.0.0'
     defaultConfig {
         applicationId 'com.smlnskgmail.jaman.hashchecker'
         minSdkVersion 21
-        targetSdkVersion 35
-        versionCode 100
-        versionName '3.7.3.1'
+        targetSdkVersion 33
+        versionCode 99
+        versionName '3.7.3'
         vectorDrawables.useSupportLibrary = true
 
         testInstrumentationRunner 'com.smlnskgmail.jaman.hashchecker.runner.AndroidJacocoTestRunner'

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -1,19 +1,15 @@
 apply plugin: 'com.android.application'
 apply plugin: 'com.jaredsburrows.license'
-apply from: 'checkstyle.gradle'
-apply from: 'cpd.gradle'
-apply from: 'jacoco.gradle'
-apply from: 'utils.gradle'
 
 android {
-    compileSdkVersion 33
-    buildToolsVersion '33.0.0'
+    compileSdkVersion 35
+    buildToolsVersion '35.0.0'
     defaultConfig {
         applicationId 'com.smlnskgmail.jaman.hashchecker'
         minSdkVersion 21
-        targetSdkVersion 33
-        versionCode 99
-        versionName '3.7.3'
+        targetSdkVersion 35
+        versionCode 100
+        versionName '3.7.3.1'
         vectorDrawables.useSupportLibrary = true
 
         testInstrumentationRunner 'com.smlnskgmail.jaman.hashchecker.runner.AndroidJacocoTestRunner'
@@ -163,8 +159,8 @@ dependencies {
     implementation 'com.j256.ormlite:ormlite-android:5.7'
 
     // Checkstyle
-    checkstyle 'com.github.fartem:checkstyle-checks-java:1.1.0'
-    checkstyle 'com.github.fartem:checkstyle-checks-android:1.2.3'
+//    checkstyle 'com.github.fartem:checkstyle-checks-java:1.1.0'
+//    checkstyle 'com.github.fartem:checkstyle-checks-android:1.2.3'
 
     // Unit tests
     testImplementation 'junit:junit:4.13.2'

--- a/app/googlePlay/release/output-metadata.json
+++ b/app/googlePlay/release/output-metadata.json
@@ -1,0 +1,20 @@
+{
+  "version": 3,
+  "artifactType": {
+    "type": "APK",
+    "kind": "Directory"
+  },
+  "applicationId": "com.smlnskgmail.jaman.hashchecker",
+  "variantName": "googlePlayRelease",
+  "elements": [
+    {
+      "type": "SINGLE",
+      "filters": [],
+      "attributes": [],
+      "versionCode": 98,
+      "versionName": "3.7.2",
+      "outputFile": "hash-checker_release.apk"
+    }
+  ],
+  "elementType": "File"
+}

--- a/app/thirdPartyStores/release/output-metadata.json
+++ b/app/thirdPartyStores/release/output-metadata.json
@@ -1,0 +1,20 @@
+{
+  "version": 3,
+  "artifactType": {
+    "type": "APK",
+    "kind": "Directory"
+  },
+  "applicationId": "com.smlnskgmail.jaman.hashchecker.thirdparty",
+  "variantName": "thirdPartyStoresRelease",
+  "elements": [
+    {
+      "type": "SINGLE",
+      "filters": [],
+      "attributes": [],
+      "versionCode": 99,
+      "versionName": "3.7.3-thirdparty",
+      "outputFile": "hash-checker_release.apk"
+    }
+  ],
+  "elementType": "File"
+}

--- a/build.gradle
+++ b/build.gradle
@@ -3,6 +3,9 @@ buildscript {
         google()
         mavenCentral()
         maven {
+            url 'https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven'
+        }
+        maven {
             url 'https://plugins.gradle.org/m2/'
         }
     }
@@ -18,6 +21,9 @@ allprojects {
     repositories {
         google()
         mavenCentral()
+        maven {
+            url 'https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven'
+        }
         maven { url "https://jitpack.io" }
     }
 }

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,6 @@
+#Thu Jul 11 15:29:55 MSK 2024
 distributionBase=GRADLE_USER_HOME
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
 distributionPath=wrapper/dists
-zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-all.zip
+zipStoreBase=GRADLE_USER_HOME

